src/state_point.F90

```diff
@@ -717,6 +724,19 @@ subroutine load_state_point()
     ! Read and overwrite random number seed
     call read_dataset(file_id, "seed", seed)
 
+    ! It is not impossible for a state point to be generated from a CE run but
+    ! to be loaded in to an MG run (or vice versa), check to prevent that.
+    call read_dataset(file_id, "run_CE", sp_run_CE)
+    if (sp_run_CE == 0) then
+      if (run_CE) &
+           call fatal_error("State point file is from multi-group run but &
+                            & current run is continous-energy!")
+    else if (sp_run_CE == 1) then
+      if (.not. run_CE) &
+           call fatal_error("State point file is from continuous-energy run but &
+                            & current run is multi-group!")
+    end if
+
```

Comment:
This is in response to the comment re: score_general being very similar (why cant i reply to it GitHub!?):

There is totally a way to wrap around the differences just like you suggest.  However, I went in with the goal that CE runtime was the most important thing to keep, especially for tallies. Putting such a wrapper will probably have a noticeable impact on runtime for something like a fully tallied HM problem.

Having said that, I am up for exploring it, but would that make more sense after this PR is merged and the distribution abstraction by @paulromano is merged as well?  That may simplify stuff in more locations of the code.